ess/src/API/EMBC.ESS/Managers/Submissions/SubmissionsManager.cs

```diff
@@ -158,9 +158,15 @@ public async Task<string> Handle(LinkRegistrantCommand cmd)
             var file = mapper.Map<Resources.Cases.EvacuationFile>(caseRecord);
             var member = file.HouseholdMembers.Where(m => m.Id == cmd.HouseholdMemberId).SingleOrDefault();
 
+            var registrantAlreadyLinked = file.HouseholdMembers.Any(m => m.LinkedRegistrantId == cmd.RegistantId);
+            if (registrantAlreadyLinked) throw new Exception($"There is already a HouseholdMember linked to this Registrant");
+
             if (member == null) throw new Exception($"HouseholdMember not found '{cmd.HouseholdMemberId}'");
 
             member.LinkedRegistrantId = cmd.RegistantId;
+            var needsAssessmentMember = file.NeedsAssessment.HouseholdMembers.Where(m => m.Id == cmd.HouseholdMemberId).SingleOrDefault();
+            if (needsAssessmentMember != null) needsAssessmentMember.LinkedRegistrantId = cmd.RegistantId;
```

Comment:
is this used anywhere?

Reply:
I don't know if it's a valid case where this member is already linked to a registrant and you are linking to a different registrant (The test does this to simplify the need to identify a unique member and registrant link)

But if you are changing the registrant link, since the repo goes through the file members and then the assessment members, if this member is in both places it will get updated at the file level (new registrant) and again at the needs assessment level (old registrant). So this resolves that issue.

Reply:
It depends which household member entity is used when updating - if the needs assessment member, then this may be valid, if the file, then it is not needed?